<a href="https://colab.research.google.com/github/financieras/curso_python/blob/main/resumen/033_input.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Input  
* Introducir datos por teclado usando `input`
* Con `input` siempre se captura un *string* aunque se escriba un número.

In [ ]:
input()     # Introduciendo un número lo toma como string. Pulse ENTER después de introducir el valor

15


'15'

In [ ]:
print("¿Cómo se llama?")
nombre = input()
print("Me alegro de conocerle", nombre)

¿Cómo se llama?
Alberto
Me alegro de conocerle Alberto


In [ ]:
input("¿Cuántos años tiene?: ")  # Añadiendo texto dentro del paréntesis

¿Cuántos años tiene?: 15


'15'

In [ ]:
edad = input("¿Cuántos años tiene?: ")  # usamos una variable para capturar el dato
print(edad)
type(edad)                              # str, es un string

¿Cuántos años tiene?: 15
15


str

In [ ]:
edad = int(input("¿Cuántos años tienes?: "))     # se pone int para convertir
print(f"Tengo {edad} años")
print(f"Y el próximo año tendré {edad+1}")      # da Error si no se pone int

¿Cuántos años tienes?: 15
Tengo 15 años
Y el próximo año tendré 16


### Convirtiendo el string en un int

In [ ]:
num = int(input("Introduzca un número entero: "))         # int convierte el string a entero
num

Introduzca un número entero: 5


5

In [ ]:
# Sumando dos enteros
num1 = int(input("Introduzca el primer número: "))
num2 = int(input("Introduzca el segundo número: "))
print("La suma es: ", num1 + num2)

Introduzca el primer número: 1
Introduzca el segundo número: 2
La suma es:  3


In [ ]:
# Concatenando dos números
num1 = input("Introduzca el primer número: ")
num2 = input("Introduzca el segundo número: ")
print("Ahora no suma, da: ", num1 + num2)                # observe que ahora no suma, concatena

Introduzca el primer número: 1
Introduzca el segundo número: 2
Ahora no suma, da:  12


In [ ]:
# Multiplicando dos enteros
num1 = int(input("Introduzca el primer número: "))
num2 = int(input("Introduzca el segundo número: "))
print("El producto de", num1 , "por", num2, "es igual a", num1*num2)
print("El producto de {} por {} es igual a {}".format(num1, num2, num1*num2))
print(f"El producto de {num1} por {num2} es igual a {num1*num2}")

Introduzca el primer número: 5
Introduzca el segundo número: 3
El producto de 5 por 3 es igual a 15
El producto de 5 por 3 es igual a 15
El producto de 5 por 3 es igual a 15


### Convirtiendo el string en un float
El separador de decimales es el punto.

In [ ]:
num = float(input("Introduzca un número con decimales: ")) # float convierte el string a flotante
num

Introduzca un número con decimales: 12.75


12.75

In [ ]:
# Dividiendo dos flotantes
num1 = float(input("Introduzca el primer número: "))
num2 = float(input("Introduzca el segundo número: "))
print("El cociente de " + str(num1) + " entre " + str(num2) + " es igual a " + str(num1 / num2))
print(f"El cociente de {num1} entre {num2} es igual a {num1/num2}")

Introduzca el primer número: 5.5
Introduzca el segundo número: 2.5
El cociente de 5.5 entre 2.5 es igual a 2.2
El cociente de 5.5 entre 2.5 es igual a 2.2


## Variable como argumento de la función input()

In [ ]:
nombre = input("Dígame su nombre: ")
apellido = input(f"Dígame su apellido, {nombre}: ")
print(f"Al final nos conocemos en persona, {nombre} {apellido}.")

Dígame su nombre: Ana
Dígame su apellido, Ana: Ruiz
Al final nos conocemos en persona, Ana Ruiz.


## Valor por defecto en un input()
Si ponemos un ```or``` podemos incluir un valor por defecto.
Si el usuario pulsa ENTER cuando se le pregunta, la repuesta será el valor por defecto.

In [ ]:
hojaCalculo = input("Indique su Hoja de Calculo favorita: ") or "Excel"
print(hojaCalculo)

Indique su Hoja de Calculo favorita: 
Excel


## Ejercicios

#### *Ejercicio 1*
* Pide por teclado dos números, primero el num1 y luego vuelve a pedir por teclado el num2.
* Finalmente imprime los dos números concatenados, no los sumes.

#### *Ejercicio 2*
1. Pregunta al usuario por el número de coches que tiene su familia.
2. Si tiene 1 coche, usando el sistema de formato `f` debe imprimir la frase:
<pre>No me gusta nada estar revisando la presión de las 4 ruedas.</pre>
3. Si tiene 2 coches debe imprimir la frase:
<pre>No me gusta nada estar revisando la presión de las 8 ruedas.</pre>
4. Al introducir el usuario el número de coches que hay en su familia la frase debe adaptarse a cada caso.

#### *Ejercicio 3*
1. Crea un programa que pida al usuario que indique un producto y su precio que contenga unos euros y unos céntimos de euro.
2. Por ejemplo, producto = "café", precio = 2.50.
3. El programa ha de imprimir la frase:
    <pre>Esta mañana he pagado 2 euros y 50 céntimos por un café.</pre>
5. Usa el sistema `f` de *format* para imprimir la frase.
6. Prueba con otros productos y precios.